dagsync/ipnisync/publisher.go

```
@@ -218,7 +223,14 @@ func (p *Publisher) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "invalid request: not a cid", http.StatusBadRequest)
 		return
 	}
-	item, err := p.lsys.Load(ipld.LinkContext{}, cidlink.Link{Cid: c}, basicnode.Prototype.Any)
+
+	ipldCtx := ipld.LinkContext{}
+	reqType := r.Header.Get(CidSchemaHeader)
+	if reqType != "" {
```

Validate and fail early if type is unknown?

---

The reasoning not to:
> The publisher, who receives the Ipni-Cid-Schema-Type HTTP header, does not validate the value, because newer values may need to be received by consumer that is using an older version of library.

---

> Consider Boost wanting to use an older version of go-libipni, 

But older versions would not set the header, right? If the header is present then we would validate. otherwise fall back on the old behaviour. 

No?

---

I mean if boost wants to use this version of go-libipni (this one being older), but both boost and indexer understand some newer hint. This way boost's older version of go-libipni can get the hint from the HTTP header, put then hint into a context and let boost extract extract that hint in its linksystem -- even though boost's go-libipni does not recognize the value of the hint.

---

OK